src/Components/homepage/Homepage.js

```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
 import Card from '../Card/Card'
 import { fetchData } from '../../Apis/Dummy-api'
 import '../Style/Homepage.css'
+import RecentlyPlayedZone from '../Recently-played-zone/Recently-played-zone'
 
 export const Homepage = () =>  {
     const [cards, setCards] = useState([])
 	const [errorMessage, setErrorMessage] = useState([])
-
+	const [recentlyPlayedCards, setRecentlyPlayedCards] = useState([])
 	const numberOfCards = 20
 
     useEffect(() => {
-        fetchData(numberOfCards).then((res) => {setCards(res)}).catch((e) => {
-            setErrorMessage(e)
+        fetchData(numberOfCards)
+        .then((res) => {setCards(res)})
+        .catch((error) => {
+            setErrorMessage(error)
             console.error('[ERROR IN HOMEPAGE COMPONENT FETCH DATA]: ' + errorMessage)
         })
     }, [])
 
+    // useEffect(() => {
+    //     if (cards.length) {
+    //         console.log(cards)
+    //         const firstCards = cards.slice(0,4)
+    //         setRecentlyPlayedCards(firstCards)
+    //     }
+    // }, [cards])
+
+    const addRecentCard = (clickedCardId) => {
+        setRecentlyPlayedCards((oldCardsArray) => {
+            const found = oldCardsArray.find((element) => element === cards[clickedCardId - 1])
+            if (found) {
+                return [...oldCardsArray]
+            }
+            else {
+                if (oldCardsArray.length >= 4) {
+                    oldCardsArray.unshift(cards[clickedCardId - 1])
+                    oldCardsArray.pop()
```

Comment:
please try avoiding mutating the array directly, in general it is a good practice to treat arrays as immutable and create new ones if data needs to be changed

you could try refactoring this with "non-mutating" methods, like `.filter`

more on this topic: https://www.sitepoint.com/immutable-array-methods-javascript/